libraries/platforms/bigblade-verilator/bsg_manycore_regression_platform.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// To make your program HammerBlade cross-platform compatible,
+// define a function with the signature of "main", and then
+// use this macro to mark it as the entry point of your program
+//
+// Example:
+//
+//    int MyMain(int argc, char *argv[]) {
+//        /* your code here */
+//    }
+//    declare_program_main("The name of your test", MyMain)
+//
+#ifdef VCS
+#define declare_program_main(test_name, name)                   \
+    int vcs_main(int argc, char *argv[]) {                      \
+        bsg_pr_test_info("Regression Test: %s\n", test_name);   \
+        int rc = name(argc, argv);                              \
+        bsg_pr_test_pass_fail(rc == HB_MC_SUCCESS);             \
```

Comment:
I feel like this might be something we should make platform independent

Reply:
what do you mean by platform independent?

Reply:
My understanding is that this is just hacking around making VCS having a main and Verilator needing a main supplied

Reply:
That understanding is correct. We could define this macro in `bsg_manycore_regression.h` if it's not defined by the platform.

Reply:
OTOH what would be the default?

Reply:
also, why is it defined for Verilator and vcs? isn't this platform just for Verilator?

Reply:
Oh yeah that's true, could get rid of the ifdef and have the .h be different in each platform